src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -1465,6 +1467,117 @@ class Simplex_tree {
     }
   }
 
+  /** \brief Retrieve good values for extended persistence, and separate the 
+   * diagrams into the ordinary, relative, extended+ and extended- subdiagrams. 
+   * Need extend_filtration to be called first!
+   * @param[in] dgm Persistence diagram obtained after calling this->extend_filtration 
+   * and this->get_persistence.
+   * @return A vector of four persistence diagrams. The first one is Ordinary, the 
+   * second one is Relative, the third one is Extended+ and the fourth one is Extended-.
+   */
+  std::vector<std::vector<std::pair<int, std::pair<double, double>>>> compute_extended_persistence_subdiagrams(const std::vector<std::pair<int, std::pair<double, double>>>& dgm){
```

Comment:
I was wondering about alternate interfaces, where this function would not be part of the Simplex_tree. Just putting them out there in case they inspire something to someone.

1. We could have something on top of the simplex tree, that presents a different interface to Persistent_cohomology. For instance, we could store minvalue and maxvalue in that object. Or Filtration_value could be something like `pair<bool,double>` and filtration(simplex) would return (true,filt) if simplex ends with maxvert, (false,filt) otherwise. This avoids modifying the filtration values, and the meaning of the normal output of persistence is not too hard to explain. We could even avoid constructing the cone explicitly, and write functions for boundary, filtration, etc that fake its existence, although we would probably need to add a second "key" field to each simplex or use a hashtable on the side to store the key of those phantom simplices.

2. We could have an Extended_persistent_cohomology, that internally uses Persistent_cohomology, and that provides various functions like relative_intervals_in_dimension, etc, already split and already with corrected values.